src/main/resources/GoogleApps/Groovy.groovy

```diff
@@ -365,27 +368,32 @@ public class GoogleAppsGroovy {
             }
         }
         String id = user.get("resourceName");
-        if(id.startsWith("people/") && id.length()>7) id = id.substring(7);
+        if(id.startsWith("people/") && id.length()>7) {
+            id = id.substring(7);
+        }
         def db = context.getDatabase()
         try {
             // Force main wiki database to create the user as global
             context.setDatabase("xwiki")
-            if(email == "")
-                email = (user.emailAddresses!=null && user.emailAddresses.size()>0) ? user.emailAddresses[0].value : "";
+            if(usersEmailAddress == "") {
+                usersEmailAddress = (user.emailAddresses!=null && user.emailAddresses.size()>0) ? user.emailAddresses[0].value : "";
```

Comment:
The test is `if(usersEmailAddress == "")` and then the `usersEmailAddress` is set again to `""`

Reply:
This should be replaced with:
```
if(usersEmailAddress == "" && user.emailAddresses!=null && user.emailAddresses.size()>0) {
  usersEmailAddress = user.emailAddresses[0].value
}
```
And maybe the test around `user.emailAddresses` could be improved even more. Will it be ever null?

Reply:
Thanks. Your reformulation looks good.
There is no documented information about the non-nullity of emailAddresses (or its cardinality).

Reply:
Reformulation commited.